### PR TITLE
Chore / Upgrade github workflow dependencies

### PR DESCRIPTION
**Pull request by langemike**
While working on PR #566 I noticed we use an outdated version for Node and our dependencies in Github Workflow-files. So I incremented the version numbers to use the latest available version.

Fixes these deprecation warnings:

> Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-node@v3, actions/upload-artifact@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.Show more
 

<br class="Apple-interchange-newline">

> [!WARNING]
> actions/upload-artifact@v3 is scheduled for deprecation on **November 30, 2024**. [Learn more.](https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/)
> Similarly, v1/v2 are scheduled for deprecation on **June 30, 2024**.
> Please update your workflow to use v4 of the artifact actions.
> This deprecation will not impact any existing versions of GitHub Enterprise Server being used by customers.

[codeql-action](https://github.com/github/) v2 (deprecated, support will end on December 5th, 2024)

### CODE REVIEW

File: .github/workflows/release-build-tag-release.yml [hunk 1]
```diff
@@ -11,11 +11,11 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [18.x]
+        node-version: ['20']
```
Comment (AntonLantukh) [inline .github/workflows/release-build-tag-release.yml]: Maybe 20.x ?

```suggestion
        node-version: [20.x]
```

Comment (ChristiaanScheermeijer) [reply .github/workflows/release-build-tag-release.yml]: Is `node-version` even consumed in this workflow? If not, we should remove the matrix, right?

File: .github/workflows/test-linting.yml [hunk 2]
```diff
@@ -8,12 +8,12 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [18.x]
+        node-version: ['20']
```
Comment (ChristiaanScheermeijer) [inline .github/workflows/test-linting.yml]: Same here, we can remove the matrix configuration and configure the `actions/setup-node` action directly:

```yml
    steps:
      - uses: actions/checkout@v4
      - name: Use Node.js
        uses: actions/setup-node@v4
        with:
          node-version: 20.x
```

This prevents the action name from changing when we update the node version 🙂

Comment (AntonLantukh) [reply .github/workflows/test-linting.yml]: @langemike I will update required checks when you move node-version definition to the `actions/setup-node@v4`.

File: .github/workflows/release-build-tag-release.yml (hunk 1 above)
Comment (langemike) [reply .github/workflows/release-build-tag-release.yml]: You are right. So I removed it.

File: .github/workflows/test-linting.yml (hunk 2 above)
Comment (langemike) [reply .github/workflows/test-linting.yml]: I made the action name static and removed the matrix configuration for `node-version` in all instances.

Comment (AntonLantukh) [reply .github/workflows/test-linting.yml]: I made changes in the required checks.

**Review by AntonLantukh — APPROVED** [review]

**Review by royschut — APPROVED** [review]

**Review by ChristiaanScheermeijer — APPROVED** [review]

**Review by MelissaDTH — APPROVED** [review]

**Pull request merged**